src/models/contact/person/index.js

```diff
@@ -50,3 +73,120 @@ export function createPersonDatasetUrl(addressBook, id = uuid()) {
 export async function getPersonAll(addressBook, fetch) {
   return getContactAll(addressBook, [PERSON_CONTACT], fetch);
 }
+
+export function getWebIdUrl(dataset, iri) {
+  const thing = getThing(dataset, iri);
+  const webIdNodeUrl = getUrl(thing, vcard.url);
+  if (webIdNodeUrl) {
+    const webIdNode = getThing(dataset, webIdNodeUrl);
+    return webIdNode && getUrl(webIdNode, vcard.value);
+  }
+  return getUrl(thing, foaf.openid);
+}
+
+export function createWebIdNode(webId, iri) {
+  const webIdNode = chain(
+    createThing(),
+    (t) => addUrl(t, rdf.type, vcardExtras("WebId")),
+    (t) => addUrl(t, vcard.value, webId)
+  );
+  const webIdNodeUrl = asUrl(webIdNode, iri);
+  return { webIdNode, webIdNodeUrl };
+}
+
+export function createPersonContact(addressBook, contact) {
+  const { containerUrl: addressBookContainerUrl } = addressBook;
+
+  const normalizedContact = {
+    emails: [],
+    addresses: [],
+    telephones: [],
+    ...contact,
+  };
+
+  const id = uuid();
+  const iri = joinPath(
+    addressBookContainerUrl,
+    PERSON_CONTAINER,
+    id,
+    INDEX_FILE
+  );
+  const { webIdNode, webIdNodeUrl } = createWebIdNode(contact.webId, iri);
+  const rootAttributeFns = getSchemaOperations(contact, webIdNodeUrl);
+  const emails = normalizedContact.emails.map(mapSchema("email"));
+  const addresses = normalizedContact.addresses.map(mapSchema("address"));
+  const telephones = normalizedContact.telephones.map(mapSchema("telephone"));
+
+  const person = defineThing(
+    { name: "this" },
+    ...[(t) => addUrl(t, rdf.type, vcard.Individual), ...rootAttributeFns],
+    ...emails.map(({ name }) => {
+      return (t) => addUrl(t, vcard.hasEmail, [iri, name].join("#"));
+    }),
+    ...addresses.map(({ name }) => {
+      return (t) => addUrl(t, vcard.hasAddress, [iri, name].join("#"));
+    }),
+    ...telephones.map(({ name }) => {
+      return (t) => addUrl(t, vcard.hasTelephone, [iri, name].join("#"));
+    })
+  );
+  const dataset = chain(
+    createSolidDataset(),
+    (d) => setThing(d, person),
+    (d) => setThing(d, webIdNode),
+    ...emails
+      .concat(addresses)
+      .concat(telephones)
+      .map(({ thing }) => (d) => setThing(d, thing))
+  );
+  return {
```

Comment:
Is there any reason why we cannot do `savePersonContact` and bundle the save logic here as well? Then the returned object should represent the newly saved person contact, using the pattern `{ dataset, thing }`.

Reply:
I see that it might conflict with the method `savePerson`. It looks this might simply be an internal function that allows us to deconstruct the complexity of `savePerson`? I would be ok keeping `createPersonContact` and with returning `{ dataset, iri }` _if_ this is an internal function, i.e. no `export` and not exposing `createContact` on the type object.